src/backend/gporca/libgpopt/include/gpopt/operators/CExpression.h

```diff
@@ -216,6 +219,18 @@ class CExpression : public CRefCount, public gpos::DbgPrintMixin<CExpression>
 		return m_cost;
 	}
 
+	void
+	SetMotionInputs(IntPtrArray *motionInputSegments)
```

Comment:
How about replacing this function with this one?
```c
	SetMotionInputs(CExpression *parent)
	{
		m_motion_input_segments = parent->m_motion_input_segments;
	}
```
Then everywhere below you can delete `->GetMotionInputs()`